plugins/bigquery/dbt/adapters/bigquery/connections.py

```diff
@@ -295,15 +297,76 @@ def drop_dataset(self, database, schema):
         dataset = self.dataset(database, schema, conn)
         client = conn.handle
 
-        with self.exception_handler('drop dataset'):
-            client.delete_dataset(
-                dataset, delete_contents=True, not_found_ok=True
-            )
+        def _drop_tables_then_dataset():
+            for table in client.list_tables(dataset):
```

Comment:
Can we use `client.delete_dataset(dataset, delete_contents=True, not_found_ok=True)` here? Check out the existing implementation - I think this might have changed in dbt since you forked the repo. We don't need to drop each table explicitly anymore :)